### PR TITLE
swssconfig: Reading multiple files in default folder automatically

### PR DESCRIPTION
**Pull request by stcheng**

### CODE REVIEW

File: swssconfig/swssconfig.cpp [hunk 1]
```diff
@@ -127,47 +133,78 @@ bool load_json_db_data(iostream &fs, vector<KeyOpFieldsValuesTuple> &db_items)
...
+    DIR *dir;
+    dirent *entry;
+    dir = opendir(path.c_str());
+    if (dir)
+    {
+        while (true)
+        {
+            entry = readdir(dir);
+            if (entry == NULL) break;
+            if (strcmp(entry->d_name, ".") && strcmp(entry->d_name, ".."))
```
Comment (kcudnik) [inline swssconfig/swssconfig.cpp]: formatting ,p lease add { } on every if, and don't inline code like that, 1 statement per line

File: swssconfig/swssconfig.cpp [hunk 2]
```diff
@@ -127,47 +133,78 @@ bool load_json_db_data(iostream &fs, vector<KeyOpFieldsValuesTuple> &db_items)
...
+        usage();
+        exit(EXIT_SUCCESS);
     }
-    catch(...)
+    else
     {
-        cout << "Failed loading json file: " << argv[1] << " Please refer to logs\n";
-        return EXIT_FAILURE;
+        for (auto i = 1; i < argc; i++)
+            files.push_back(string(argv[i]));
```
Comment (kcudnik) [inline swssconfig/swssconfig.cpp]: brackets

**Comment by stcheng** [comment]
Updated

File: swssconfig/swssconfig.cpp [hunk 3]
```diff
@@ -127,47 +133,78 @@ bool load_json_db_data(iostream &fs, vector<KeyOpFieldsValuesTuple> &db_items)
...
+
+            if (!write_db_data(db_items))
+            {
+                SWSS_LOG_ERROR("Failed applying data from JSON file:%S", i.c_str());
+                return EXIT_FAILURE;
+            }
+        }
+        catch(const exception &e)
+        {
+            if (fs.is_open()) fs.close();
```
Comment (stcheng) [inline swssconfig/swssconfig.cpp]: resolved

File: swssconfig/swssconfig.cpp (hunk 2 above)
Comment (stcheng) [inline swssconfig/swssconfig.cpp]: resolved

File: swssconfig/swssconfig.cpp (hunk 1 above)
Comment (stcheng) [inline swssconfig/swssconfig.cpp]: resolved

**Comment by kcudnik** [comment]
ack

File: swssconfig/swssconfig.cpp [hunk 4]
```diff
@@ -1,8 +1,12 @@
 #include <stdlib.h>
+#include <dirent.h>
+
+#include <cstring>
```
Comment (qiluo-msft) [inline swssconfig/swssconfig.cpp]: why not use <string.h>, as other c-header file?

File: swssconfig/swssconfig.cpp [hunk 5]
```diff
@@ -127,47 +134,88 @@ bool load_json_db_data(iostream &fs, vector<KeyOpFieldsValuesTuple> &db_items)
...
     }
 
-    try
+    for (auto i : files)
     {
-        if (!write_db_data(db_items))
+        SWSS_LOG_NOTICE("Loading config from JSON file:%s...", i.c_str());
+
+        vector<KeyOpFieldsValuesTuple> db_items;
+        fstream fs;
```
Comment (qiluo-msft) [inline swssconfig/swssconfig.cpp]: ifstream?

Comment (qiluo-msft) [inline swssconfig/swssconfig.cpp]: if not reuse fs in loop, you can remove all the fs.close()

File: swssconfig/swssconfig.cpp (hunk 4 above)
Comment (stcheng) [inline swssconfig/swssconfig.cpp]: i will use string.h. wondering when cstring will be used.

File: swssconfig/swssconfig.cpp (hunk 5 above)
Comment (stcheng) [inline swssconfig/swssconfig.cpp]: resolved

File: swssconfig/swssconfig.cpp [hunk 6]
```diff
@@ -127,47 +134,88 @@ bool load_json_db_data(iostream &fs, vector<KeyOpFieldsValuesTuple> &db_items)
...
+{
+    vector<string> ret;
+    DIR *dir;
+    dirent *entry;
+    dir = opendir(path.c_str());
+    if (dir)
+    {
+        while (true)
+        {
+            entry = readdir(dir);
```
Comment (stcheng) [inline swssconfig/swssconfig.cpp]: resolved

File: swssconfig/swssconfig.cpp (hunk 5 above)
Comment (qiluo-msft) [inline swssconfig/swssconfig.cpp]: I mean moving fs variable inside the loop block, not reuse it. ref: http://stackoverflow.com/questions/2284775/c-can-i-reuse-fstream-to-open-and-write-multiple-files

File: swssconfig/swssconfig.cpp [hunk 7]
```diff
@@ -127,47 +134,88 @@ bool load_json_db_data(iostream &fs, vector<KeyOpFieldsValuesTuple> &db_items)
...
+            }
+            fs.close();
+
+            if (!write_db_data(db_items))
+            {
+                SWSS_LOG_ERROR("Failed applying data from JSON file:%S", i.c_str());
+                return EXIT_FAILURE;
+            }
+        }
+        catch(const exception &e)
```
Comment (qiluo-msft) [inline swssconfig/swssconfig.cpp]: Not applied to latest code. You may ignore.

File: swssconfig/swssconfig.cpp (hunk 5 above)
Comment (qiluo-msft) [inline swssconfig/swssconfig.cpp]: I mean move the variable into loop block, and not to reuse it. ref: http://stackoverflow.com/questions/2284775/c-can-i-reuse-fstream-to-open-and-write-multiple-files

File: swssconfig/swssconfig.cpp (hunk 7 above)
Comment (qiluo-msft) [inline swssconfig/swssconfig.cpp]: Not applied to latest code. You may ignore.

File: swssconfig/swssconfig.cpp (hunk 5 above)
Comment (stcheng) [inline swssconfig/swssconfig.cpp]: resolved

File: swssconfig/swssconfig.cpp [hunk 8]
```diff
@@ -17,9 +21,12 @@ const char* const op_name            = "OP";
 const char* const name_delimiter     = ":";
 const int el_count = 2;
 
-void usage(char **argv)
+const string SWSS_CONFIG_DIR    = "/etc/swss/config.d/";
+
+void usage()
 {
-    cout << "Usage: " << argv[0] << " json_file_path\n";
+    cout << "Usage: swssconfig [FILE...]" << endl;
```
Comment (stcheng) [inline swssconfig/swssconfig.cpp]: will update this later to use getopt()

File: swssconfig/swssconfig.cpp [hunk 9]
```diff
@@ -127,47 +134,88 @@ bool load_json_db_data(iostream &fs, vector<KeyOpFieldsValuesTuple> &db_items)
     return true;
 }
 
+vector<string> read_directory(const string &path)
```
Comment (stcheng) [inline swssconfig/swssconfig.cpp]: update later

**Pull request merged**